boefjes/boefjes/plugins/kat_nmap_tcp/main.py

```diff
@@ -1,42 +1,21 @@
+import os
+import subprocess
 from ipaddress import IPv6Address, ip_address
-from os import getenv
 
-import docker
-
-from boefjes.job_models import BoefjeMeta
-
-NMAP_IMAGE = "instrumentisto/nmap:latest"
-TOP_PORTS_MAX = 65535
 TOP_PORTS_DEFAULT = 250
-TOP_PORTS_MIN = 1
 
 
-def run_nmap(args: list[str]) -> str:
-    """Run Nmap in Docker."""
-    client = docker.from_env()
-    return client.containers.run(NMAP_IMAGE, args, remove=True).decode()
+def run(boefje_meta: dict):
+    top_ports_key = "TOP_PORTS"
```

Comment:
Ideally we'd use just the same env-var, and specify the env vars per boefje in the katalogus instead of trying the boefje-name inside the boefje itself.

Reply:
Indeed, but also see the considerations in the issue with respect to backward compatibility and the fact that we do not have a UI to configure the arguments. We discussed last week that we can move forward with this code/approach and now start a more proper Design refinement flow as this ticket is supposed to kick off: #2828